### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by Alchav**
Correctly set item classifications.
Remove some useless code.
Set one bomb to progression.
Delete unused Rules.py
Create filler item if starting weapon subtracted from item_amounts.

### CODE REVIEW

**Comment by t3hf1gm3nt** [comment]
could you not delete rules.py? i have plans for it. otherwise should be good to try and merge

**Comment by Alchav** [comment]
The Rules file here is just a copy of generic/Rules.py, is there anything in it you plan to use that could not be accomplished from using the generic rules file?

**Comment by t3hf1gm3nt** [comment]
it was more i planned on moving the add_rules from init over to rules.py at some point. and also make some more defined rules besides just has x item

**Comment by Alchav** [comment]
Reminder with this PR there is a deficit of 1 item. I don't think adding a filler item when placing a starting weapon is the right solution, as that takes an item out of the pool but also takes a location out of the location pool, so that shouldn't need to have a replacement. I think the item counts should just be 1 higher. but I wouldn't know what to change there.

**Comment by t3hf1gm3nt** [comment]
maybe revert what you did when it comes to the starting weapon then and i'll try to figure out what im gonna do about the item pool. so basically it will be the classification fixes and the get filler item definition left i think? as well as removing the unused get rule on region code

**Comment by Alchav** [comment]
Using the code as is and incrementing an item's amount by 1 should fix it. Reverting it would return to the item pool size differing depending on the chosen starting_weapon

**Comment by t3hf1gm3nt** [comment]
actually wait: did you test with only ExpandedPool = true and found a deficit, or is it also short one for ExpandedPool = false? because if its just short one for true then add a five rupees to item_amounts_all in items.py and see what happens. i had taken one out right before my test because it was saying there was one too many. if its the latter, i'll probably just merge this as is and try to work it out from there.

**Comment by Alchav** [comment]
I had only tested with it on, I was generating with a blank yaml. however, having it set to false seems to also result in a deficit. It also keeps failing generation as it cannot place all advancement items.

File: worlds/tloz/__init__.py [hunk 1]
```diff
@@ -165,9 +166,10 @@ def create_items(self):
                 if item_name == starting_weapon:  # To remove the extra Sword if that's our weapon
                     item_amounts[item_name] -= 1
                 if self.multiworld.TriforceLocations[self.player] > 0 or item_name != "Triforce Fragment":
-                    i = 0
                     for i in range(0, item_amounts[item_name]):
                         self.multiworld.itempool.append(self.create_item(item_name))
+                    if item_name == "Bomb":
+                        self.multiworld.itempool[-1].classification = ItemClassification.progression
```
Comment (t3hf1gm3nt) [inline worlds/tloz/__init__.py:172]: correct me if im wrong, but wouldn't this make EVERY bomb progression? i know we need to mark at least one bomb as progression for logic, so we should probably mark the one that gets placed in a shop up at the beginning of this function as such then. that might be the cause of the "cant place all advancement items" issue

Comment (Alchav) [reply worlds/tloz/__init__.py:172]: [-1] accesses the last item in a list. This is done after all the bombs are added, and only modifies the last one added

Comment (t3hf1gm3nt) [reply worlds/tloz/__init__.py:172]: ok gotcha

**Comment by t3hf1gm3nt** [comment]
cant replicate the deficit without this fix, but can replicate the cant place all advancement with JUST the line change that fixes items getting created with the right classification. so that seems to tell me that something is wrong with how TLOZ's items are classified. too many progression items to be able to fit into the smaller pool with false ExpandedPool. the few test gens i tried to get couldn't place 9 items. guess what we have exactly 9 of in the standard item pool? Heart Containers. So might also be an issue with rules being set up to guarantee a certain amount of hearts to get into a particular dungeon

**Comment by t3hf1gm3nt** [comment]
ooooooh i bet i know why. its because of how i changed take anys to work in false ExpandedPool: AP doesn't see the extra 4 hearts you could possibly get by going to the take any caves since i dont make AP locations for them anymore with preplaced hearts, so it struggling with the dungeon rules around needing a certain amount of hearts

**Comment by t3hf1gm3nt** [comment]
ok im dumb when it comes to git and how to get an update pushed to your PR so you can test. i pushed a commit to my own branch that maybe you can cherry pick or something https://github.com/t3hf1gm3nt/Archipelago/commit/c0b7baeb7010ded183cad63b824a482458871223 should fix the cannot place all advancement errors

**Comment by Alchav** [comment]
PR as-is does not fail generation, but still has a one item deficit with expanded pool off

**Comment by Alchav** [comment]
item_amounts_all five rupees was incremented by 1

**Comment by t3hf1gm3nt** [comment]
yeah in messing with things i realize i need to completely overhaul how the item pool is genned and pre-placed items are placed, but that's outside the scope of your PR. i can take it from here so I'll merge this in. thanks for the help!

**Pull request merged**